src/qinterface/qinterface.cpp

```diff
@@ -168,39 +187,68 @@ void QInterface::CZ(bitLenInt control, bitLenInt target, bitLenInt length)
     ControlledLoopFixture(length, [&](bitLenInt bit) { CZ(control + bit, target + bit); });
 }
 
-/// "AND" compare a bit range in QInterface with a classical unsigned integer, and store result in range starting at
-/// output
-void QInterface::CLAND(bitLenInt qInputStart, bitCapInt classicalInput, bitLenInt outputStart, bitLenInt length)
+/// Apply controlled S gate to each bit
+void QInterface::CS(bitLenInt control, bitLenInt target, bitLenInt length)
 {
-    bool cBit;
-    for (bitLenInt i = 0; i < length; i++) {
-        cBit = bitSlice(i, classicalInput);
-        CLAND(qInputStart + i, cBit, outputStart + i);
-    }
+    ControlledLoopFixture(length, [&](bitLenInt bit) { CS(control + bit, target + bit); });
 }
 
-/// "OR" compare a bit range in QInterface with a classical unsigned integer, and store result in range starting at
-/// output
-void QInterface::CLOR(bitLenInt qInputStart, bitCapInt classicalInput, bitLenInt outputStart, bitLenInt length)
+/// Apply controlled IS gate to each bit
+void QInterface::CIS(bitLenInt control, bitLenInt target, bitLenInt length)
 {
-    bool cBit;
-    for (bitLenInt i = 0; i < length; i++) {
-        cBit = bitSlice(i, classicalInput);
-        CLOR(qInputStart + i, cBit, outputStart + i);
+    ControlledLoopFixture(length, [&](bitLenInt bit) { CIS(control + bit, target + bit); });
+}
+
+/// Apply controlled T gate to each bit
+void QInterface::CT(bitLenInt control, bitLenInt target, bitLenInt length)
+{
+    ControlledLoopFixture(length, [&](bitLenInt bit) { CT(control + bit, target + bit); });
+}
+
+/// Apply controlled IT gate to each bit
+void QInterface::CIT(bitLenInt control, bitLenInt target, bitLenInt length)
+{
+    ControlledLoopFixture(length, [&](bitLenInt bit) { CIT(control + bit, target + bit); });
+}
+
+/// Apply controlled "PhaseRootN" gate to each bit
+void QInterface::CPhaseRootN(bitLenInt n, bitLenInt control, bitLenInt target, bitLenInt length)
+{
+    if (n == 0) {
+        return;
+    } else if (n == 1) {
+        CZ(control, target, length);
+        return;
     }
+
+    ControlledLoopFixture(length, [&](bitLenInt bit) { CPhaseRootN(n, control + bit, target + bit); });
 }
```

Comment:
This looks like another idiom that might be worth extracting, since I see it repeated several times.  Either with or without the special cases for `n == 0` and `n == 1`.

Reply:
`ControlledLoopFixture()` cases are slightly irregular. I considered this already, but I'll take another look and see what's appropriate.